nixos/modules/services/computing/slurm/slurm.nix

```diff
@@ -369,13 +381,21 @@ in
       wants = [ "network-online.target" ];
 
       serviceConfig = {
-        Type = "forking";
-        KillMode = "process";
-        ExecStart = "${wrappedSlurm}/bin/slurmd";
-        PIDFile = "/run/slurmd.pid";
+        Delegate="Yes";
         ExecReload = "${pkgs.coreutils}/bin/kill -HUP $MAINPID";
+        # The 23.02.06.01 release assumes slurm*d run as Type = "simple",
+        # with the `-D` flag ensuring the daemons stay in foreground.
+        # Already in 23.11.0.0-rc1 the template .service files
+        # suggest to use Type = "notify", and introduce the `--systemd`
+        # flag instead of `-D`. We'll want to update this line with the release:
+        ExecStart = "${wrappedSlurm}/bin/slurmd -D";
+        KillMode = "process";
         LimitMEMLOCK = "infinity";
-        Delegate="Yes";
+        LimitNOFILE = 131072;
+        LimitSTACK = "infinity";
+        RuntimeDirectoryMode = "0755";
+        RuntimeDirectory = "slurm";
```

Comment:
Actually I got ahead of myself. Upstream doesn't use `RuntimeDirectory` in 23.02.06 yet: https://github.com/SchedMD/slurm/blob/slurm-23-02-6-1/etc/slurmd.service.in

It does in the next release: https://github.com/SchedMD/slurm/blob/6ed43eeee266016ebae370668937ce04901367da/etc/slurmd.service.in#L11-L12

Same goes for slurmdbd by the way

Reply:
The tests pass 🤔